### PR TITLE
core-edge: commit in batches on edge machines

### PR DESCRIPTION
**Pull request by martinfurmanski**
Includes a major overhaul of the two major classes
which do the polling and the transaction applying.

### CODE REVIEW

File: enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/BatchingTxApplier.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+     * @param tx The transaction to be queued for application.
+     */
+    public void queue( CommittedTransactionRepresentation tx )
+    {
+        long receivedTxId = tx.getCommitEntry().getTxId();
+        long expectedId = lastQueuedTxId + 1;
+
+        if ( receivedTxId != expectedId )
+        {
+            log.warn( "Out of order txId: " + receivedTxId );
```
Comment (mneedham) [inline enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/BatchingTxApplier.java]: Doesn't need to be there for the merge but we should log `expectedId` as well so we have more info on what's gone wrong.

File: enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java [hunk 2]
```diff
@@ -19,72 +19,181 @@
...
+    enum Timeouts implements TimeoutName
+    {
+        TX_PULLER_TIMEOUT
+    }
+
+    interface Handler
+    {
+        default void onTimeout( TxPollingClient ctx ) {}
+        default void onTxReceived( TxPollingClient ctx, TxPullResponse tx ) {}
+        default void onTxStreamingComplete( TxPollingClient ctx ) {}
```
Comment (mneedham) [inline enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java:60]: Shouldn't we log an error as default behaviour?

Comment (martinfurmanski) [inline enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java]: No, this is an interface which also defines the default behaviour of not doing anything for simplicity.

Comment (martinfurmanski) [inline enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java]: No, this is an interface which also defines the default behaviour of not doing anything for simplicity.

**Comment by mneedham** [comment]
Looks good otherwise

File: enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java (hunk 2 above)
Comment (martinfurmanski) [inline enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/tx/edge/TxPollingClient.java:60]: I'll see what I can do.

**Pull request merged**